sched/semaphore/sem_reset.c

```diff
@@ -76,38 +81,63 @@ int nxsem_reset(FAR sem_t *sem, int16_t count)
 
   flags = enter_critical_section();
 
-  /* A negative count indicates that the negated number of threads are
-   * waiting to take a count from the semaphore.  Loop here, handing
-   * out counts to any waiting threads.
-   */
-
-  while (atomic_read(NXSEM_COUNT(sem)) < 0 && count > 0)
+  if (NXSEM_IS_MUTEX(sem))
```

Comment:
A mutex does not have a count, what is the use case for nxmutex_reset ?

Reply:
https://github.com/apache/nuttx/blob/8832136b69865d54124c9a48d3a87c29ecca152c/drivers/serial/serial.c#L2253